vcd/resource_vcd_nsxt_route_advertisement_test.go

```diff
@@ -0,0 +1,177 @@
+//go:build network || nsxt || ALL || functional
+// +build network nsxt ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"regexp"
+	"strconv"
+	"testing"
+)
+
+func TestAccVcdNsxtRouteAdvertisement(t *testing.T) {
+	preTestChecks(t)
+	skipNoNsxtConfiguration(t)
+
+	isRouteAdvertisementEnable := true
```

Comment:
It would be too luxurious to skip this test always as our environments do not have a dedicated Edge Gateway yet. The best here would be to invoke `govcd` and temporarily change Edge Gateway to Dedicated with a `defer` function to set it back so that other tests don't brake.

Something similar to https://github.com/vmware/terraform-provider-vcd/blob/main/vcd/resource_vcd_nsxt_alb_pool_test.go#L780-L791